packages/dht/src/dht/DhtNode.ts

```diff
@@ -550,11 +550,10 @@ export class DhtNode extends EventEmitter<Events> implements ITransport {
     }
 
     public async storeDataToDht(key: DhtAddress, data: Any, creator?: DhtAddress): Promise<PeerDescriptor[]> {
-        const connectedEntryPoints = this.getConnectedEntryPoints()
-        if (this.peerDiscovery!.isJoinOngoing() && connectedEntryPoints.length > 0) {
-            return this.storeDataToDhtViaPeer(key, data, sample(connectedEntryPoints)!)
-        }
-        return this.storeManager!.storeDataToDht(key, data, creator ?? this.getNodeId())
+        return this.executeRecursiveOperation(
+            () => this.storeManager!.storeDataToDht(key, data, creator ?? this.getNodeId()),
+            (connectedEntryPoint) => this.storeDataToDhtViaPeer(key, data, connectedEntryPoint)
```

Comment:
The creator argument is not used in `this.storeDataToDhtViaPeer(key, data, connectedEntryPoint)`? Maybe we should remove that parameter from the public API?

Reply:
Added this ti future improvements